internal/provider/provider.go

```diff
@@ -760,6 +760,8 @@ func Provider() *schema.Provider {
 			"aws_neptune_engine_version":        neptune.DataSourceEngineVersion(),
 			"aws_neptune_orderable_db_instance": neptune.DataSourceOrderableDBInstance(),
 
+			"aws_networkfirewall_firewall": networkfirewall.DataSourceFirewall(),
```

Comment:
Note that you might need to rebase since there are now other data sources for `aws_networkfirewall` in the latest release: https://github.com/hashicorp/terraform-provider-aws/releases/tag/v4.23.0